File: problems/0836-rectangle-overlap/solution.ex
defmodule Solution do
  @spec is_rectangle_overlap(rec1 :: [integer], rec2 :: [integer]) :: boolean
  def is_rectangle_overlap([x1, y1, x2, y2], [x3, y3, x4, y4]) do
    x1 < x4 and x3 < x2 and y1 < y4 and y3 < y2
  end
end

File: problems/0836-rectangle-overlap/solution.erl
-module(solution).
-export([is_rectangle_overlap/2]).

-spec is_rectangle_overlap(Rec1 :: [integer()], Rec2 :: [integer()]) -> boolean().
is_rectangle_overlap([X1, Y1, X2, Y2], [X3, Y3, X4, Y4]) ->
    X1 < X4 andalso X3 < X2 andalso
    Y1 < Y4 andalso Y3 < Y2.

